### PR TITLE
[Core/Win] Add type icon support

### LINKED ISSUE #118: Type icon provider
VS displays icons for the various control types, we need an API for requesting icons for selected objects.
untitled

### PR DESCRIPTION
**Pull request by ermau**
Closes #118

### CODE REVIEW

**Comment by CartBlanche** [comment]
The `TargetPlatform.IconProvider` is null. Are the changes supposed just to be back-end or is the visual part missing?

File: Xamarin.PropertyEditing/ViewModels/PanelViewModel.cs [hunk 1]
```diff
@@ -88,6 +90,14 @@ public void SetIsExpanded (string group, bool isExpanded)
 			SetIsExpanded (ArrangeMode, group, isExpanded);
 		}
 
+		public Task<Stream> GetIconAsync ()
+		{
+			if (TargetPlatform.IconProvider == null)
```
Comment (CartBlanche) [inline Xamarin.PropertyEditing/ViewModels/PanelViewModel.cs]: The `TargetPlatform.IconProvider` is never set, so always null. Are the changes supposed just to be back-end or is the visual part missing?

Comment (ermau) [reply Xamarin.PropertyEditing/ViewModels/PanelViewModel.cs]: There's no test icon provider in the standalone, this is just the support for it.

Comment (CartBlanche) [reply Xamarin.PropertyEditing/ViewModels/PanelViewModel.cs]: @ermau will a test provider be added to the Stand Alone in a future PR and will a test provider  be needed for Mac, down the line?

Comment (ermau) [reply Xamarin.PropertyEditing/ViewModels/PanelViewModel.cs]: Yes

**Pull request merged**